src/urllib3/util/url.py

```diff
@@ -210,7 +210,7 @@ def _encode_invalid_chars(component, allowed_chars, encoding="utf-8"):
     if component is None:
         return component
 
-    component = six.ensure_text(component)
+    component = ensure_str(component)
```

Comment:
This call to `to_str()` would need an `encoding="utf-8"` if implemented like below.

Reply:
> would need an encoding="utf-8" if implemented like below.

Do you mean `encoding=encoding`? because the `_encode_invalid_chars` accepts an `encoding` param with default value to `"utf-8"`

Reply:
Hm, that parameter has been doing basically nothing it seems. You can remove the parameter and at the end of the `_encode_invalid_chars()` change `encoded_component.decode(encoding)` to `.decode()` since percent-encoding implies the end result is ASCII.